wgsl/index.bs

```diff
@@ -5120,30 +5120,104 @@ If the `sampler_comparison` uses bilinear filtering then the returned value is
 the filtered average of these values, otherwise the comparision result of a
 single texel is returned.
 
-**TODO:**
 
-<pre class='def'>
-`vec4<type> textureLoad(texture_storage_ro_1d      , i32       coords)`
-`vec4<type> textureLoad(texture_storage_ro_2d      , vec2<i32> coords)`
-`vec4<type> textureLoad(texture_storage_ro_1d_array, vec2<i32> coords)`
-`vec4<type> textureLoad(texture_storage_ro_3d      , vec3<i32> coords)`
-`vec4<type> textureLoad(texture_storage_ro_2d_array, vec3<i32> coords)`
-  %32 = OpImageRead %v4float %texture_storage_ro %coords
+### `textureLoad` ### {#textureload}
+
+Reads a single texel from a texture without sampling or filtering.
+
+<!-- TODO(ben-clayton): Remove parameter alignment padding -->
+
+```rust
+textureLoad(t : texture_1d<T>,                    coords : i32)                                                           -> vec4<T>
+textureLoad(t : texture_1d_array<T>,              coords : i32,       array_index : i32)                                  -> vec4<T>
+textureLoad(t : texture_2d<T>,                    coords : vec2<i32>)                                                     -> vec4<T>
+textureLoad(t : texture_2d<T>,                    coords : vec2<i32>,                    level : i32)                     -> vec4<T>
+textureLoad(t : texture_2d_array<T>,              coords : vec2<i32>, array_index : i32)                                  -> vec4<T>
+textureLoad(t : texture_2d_array<T>,              coords : vec2<i32>, array_index : i32, level : i32)                     -> vec4<T>
+textureLoad(t : texture_3d<T>,                    coords : vec3<i32>)                                                     -> vec4<T>
+textureLoad(t : texture_3d<T>,                    coords : vec3<i32>,                    level : i32)                     -> vec4<T>
+textureLoad(t : texture_multisampled_2d<T>,       coords : vec2<i32>,                                 sample_index : i32) -> vec4<T>
+textureLoad(t : texture_multisampled_2d_array<T>, coords : vec2<i32>, array_index : i32,              sample_index : i32) -> vec4<T>
+textureLoad(t : texture_depth_2d,                 coords : vec2<i32>)                                                     -> f32
+textureLoad(t : texture_depth_2d,                 coords : vec2<i32>,                    level : i32)                     -> f32
```

Comment:
While I was implementing these in tint, it occurred to me that it's slightly odd that we have `textureSampleLevel()`, but we mix level and non-level overloads in `textureLoad()`.
Should we move these overloads to a ``textureLoadLevel()` method to be consistent?

Apart from the naming inconsistency, I guess it's possible that keeping these in `textureLoad` may cause overload collisions in the future.

Reply:
Let's talk about this on the call. I don't have a preference, but a point can be made about `textureSampleLevel` having a very different semantic from just `textureSample` since there is no more derivatives involved. I.e. it can be used in a non-uniform context.

Reply:
Decided to just overload `textureLoad`